wikipedia/track.py

```diff
@@ -73,16 +74,20 @@ def __init__(self, track, params, **kwargs):
         self.search_application_params = SearchApplicationParams(track, params)
 
     def params(self):
-        query = next(self._queries_iterator)
-        return {
-            "method": "POST",
-            "path": f"{SEARCH_APPLICATION_ROOT_ENDPOINT}/{self.search_application_params.name}/_search",
-            "body": {
-                "params": {
-                    "query_string": query,
+        try:
+            query = next(self._queries_iterator)
+            return {
+                "method": "POST",
+                "path": f"{SEARCH_APPLICATION_ROOT_ENDPOINT}/{self.search_application_params.name}/_search",
+                "body": {
+                    "params": {
+                        "query_string": query,
+                    },
                 },
-            },
-        }
+            }
+        except StopIteration:
```

Comment:
**Non-blocking question**: Do we see any exception for this part? Does this happen due to long running of the benchmark?